### PR TITLE
feat(frontend): dynamic RS line that floats in the price chart's empty space

### PR DESCRIPTION
**Pull request by xang1234**
## Summary

Follow-up to #215 (which moved the RS line into a fixed strip below the price). That strip auto-scaled the RS line into ~12% of height, so it looked **flat and unreadable**. This makes the RS line **dynamic**: it floats in the price chart's empty space, growing to use available room and shrinking to a safe floor strip on price/RS divergence — and **never overlaps the candles**. Its peak can now rise above the lowest price bar (the original goal). Still one chart, one time axis.

## How it works

- **`rsBand.js`** (new, pure, unit-tested) — `computeRsBand()` finds the tallest overlap-free RS scale margin for the visible window (logarithmic price scale, linear RS scale), clamped to a **12%–38%** band; `rsBandForRange()` aligns RS values to candles by time and filters to the visible range.
- **`CandlestickChart.jsx`** — applies the computed margin to the `rs` overlay scale and **recomputes on data change + pan/zoom** (debounced 80ms, since price re-auto-scales to the visible window). The "RS" label tracks the live band top. Price reclaims full height when the RS line is hidden (toggle off / weekly).

## Maintainability (decomposition)

This work would have pushed `CandlestickChart.jsx` over 1000 lines, so the file was decomposed along the way (all behavior-preserving):
- `ChartSkeleton.jsx` — loading skeleton extracted
- `candlestickData.js` — pure OHLCV→series transforms + EMA extracted
- `createPriceChartSeries.js` — chart + 6-series construction factory extracted
- RS strip-layout branches deduped

Net: `CandlestickChart.jsx` **965 → 719 lines**; the init effect **177 → ~90 lines**.

## Verification

- **Unit tests** (`rsBand.test.js`, 11): correlated uptrend → tall band + no-overlap property; divergence → floor strip; flat / <2 bars / non-positive price → floor; cap + min respected; gap-margin met in non-degraded cases; time-range filtering.
- **Visual (Playwright, static-mode harness)**: correlated → RS rises into the empty space (peak above the lowest bar), divergence → degrades to the floor strip, separation stress test (extreme candles, RS at band edges, volume spike) holds at full and small heights, and the chart renders identically after each decomposition.

## Test Plan
- [ ] `cd frontend && npm run test:run` — rsBand suite green
- [ ] Open a stock chart, confirm the orange RS line is dynamic below the candles and never overlaps them; toggle RS off → price reclaims the space; switch to weekly → RS hides.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * RS line now dynamically repositions within the chart’s empty space to avoid overlapping candlesticks; RS label tracks the live band top.
  * Added an animated chart loading skeleton for a nicer initial load experience.
  * Improved chart rendering with consolidated price/volume/indicator series for smoother updates.

* **Tests**
  * Added comprehensive unit tests validating RS band positioning across edge cases.

* **Documentation**
  * Added implementation plan and design spec for dynamic RS band behavior.

### CODE REVIEW

**Comment by xang1234** [comment]
@coderabbitai review

File: frontend/src/components/Charts/candlestickData.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+  const weeklyData = [];
+  let currentWeek = null;
+
+  dailyData.forEach((day) => {
+    const date = new Date(day.date);
+    const weekStart = new Date(date);
+    weekStart.setDate(date.getDate() - date.getDay()); // Start of week (Sunday)
+    const weekKey = weekStart.toISOString().split('T')[0];
+
```
[the remark this answers is not in the document]
Comment (xang1234) [reply frontend/src/components/Charts/candlestickData.js]: Fixed in e4406584. Confirmed the shift: under `America/Los_Angeles` the local-time bucketing put a Sunday `2025-01-05` into the `2024-12-30` week (off by a full week); switched to `getUTCDay`/`getUTCDate`/`setUTCDate` plus an explicit `T00:00:00Z`, now stable across UTC/LA/Tokyo. Pre-existing logic moved verbatim into this file in the refactor; fixed in passing.

File: docs/superpowers/plans/2026-06-05-rs-line-dynamic-empty-space.md [hunk 2]
```diff
@@ -0,0 +1,585 @@
...
+```bash
+cd /Users/admin/Documents/Work/stock-screener/.claude/worktrees/fix+rs-line-below-price
+git push
+```
+
+- [ ] **Step 2: Update PR #215 description** so it reflects the dynamic band (superseding the fixed strip):
+
+```bash
+gh pr edit 215 --body "$(cat <<'EOF'
+## Summary
```
[the remark this answers is not in the document]
Comment (xang1234) [reply docs/superpowers/plans/2026-06-05-rs-line-dynamic-empty-space.md]: Fixed in 32c9edb9 — the plan's Step 2 now targets #216.

**Comment by xang1234** [comment]
Re the outside-diff note on `keepPreviousData: true` (CandlestickChart.jsx:94): verified we're on `@tanstack/react-query@5.90.12`, where that query option is indeed silently ignored. Declining to change it in this PR because (a) it's pre-existing and outside this PR's diff, and (b) the same query already sets `placeholderData: getCachedData`, so the placeholder/keep-previous behavior is already handled and the dead option is a no-op. Reasonable as a trivial standalone cleanup (drop the dead key), but out of scope here.
